src/modules/rlm_magic/rlm_magic.c

```diff
@@ -0,0 +1,249 @@
+/*
+ *   This program is is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or (at
+ *   your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, write to the Free Software
+ *   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/**
+ * $Id$
+ * @file rlm_magic.c
+ * @brief Example module code.
+ *
+ * @copyright 2013 The FreeRADIUS server project
+ * @copyright 2013 your name (your name\@address)
+ */
+RCSID("$Id$")
+
+#include <freeradius-devel/radiusd.h>
+#include <freeradius-devel/modules.h>
+#include <freeradius-devel/rad_assert.h>
+
+/*
+ *	Define a structure for our module configuration.
+ *
+ *	These variables do not need to be in a structure, but it's
+ *	a lot cleaner to do so, and a pointer to the structure can
+ *	be used as the instance handle.
+ */
+typedef struct
+{
+	unsigned short port;
+	char const *socket_address;
+	bool is_local;
+} rlm_magic_t;
+
+/*
+ *	A mapping of configuration file names to internal variables.
+ */
+static const CONF_PARSER module_config[] = {
+	{"port", FR_CONF_OFFSET(PW_TYPE_SHORT, rlm_magic_t, port), "12345"},
+	{"is_local", FR_CONF_OFFSET(PW_TYPE_BOOLEAN, rlm_magic_t, is_local), false},
+	{"socket-address", FR_CONF_OFFSET(PW_TYPE_STRING, rlm_magic_t, socket_address), "127.0.0.1"},
+	CONF_PARSER_TERMINATOR};
+
+/*
+ *	Do any per-module initialization that is separate to each
+ *	configured instance of the module.  e.g. set up connections
+ *	to external databases, read configuration files, set up
+ *	dictionary entries, etc.
+ */
+static int mod_instantiate(UNUSED CONF_SECTION *conf, void *instance)
+{
+	rlm_magic_t *inst = instance;
+	DEBUG2("Creating Magic module");
+
+	if (inst->is_local)
+	{
+		#ifndef HAVE_SYS_UN_H
+			ERROR("Trying to use local unix sockets on a system that does not support it");
+			return -1;
+		#endif
+		DEBUG2("Using local Unix socket %s for authentication", inst->socket_address);
+	}
+	else
+	{
+		DEBUG2("Using remote server %s:%i for authentication", inst->socket_address, inst->port);
+	}
+
+	return 0;
+}
+
+/*
+ *	Don't currently do anything here
```

Comment:
Just omit functions which don't do anything.

Reply:
is it safe to leave them in if at some point it is likely to do something? I suppose its not really more or less work but was wondering if there was some adverse reason for leaving them there